src/libraries/System.Text.RegularExpressions/tests/RegexCultureTests.cs

```diff
@@ -119,5 +141,137 @@ Regex[] Create(string input, CultureInfo info, RegexOptions additional)
+                yield return new object[] { "(?i:iI+)", options, invariant, "abcIIIxyz", "III" };
+                yield return new object[] { "(?i:iI+)", options, invariant, "abc\u0130\u0131Ixyz", "" };
+
+                // Expected answers in the Turkish culture
+                yield return new object[] { "(?i:I)", options, turkish, "xy\u0131ab", "\u0131" };
+                yield return new object[] { "(?i:iI+)", options, turkish, "abcIIIxyz", "" };
+                yield return new object[] { "(?i:iI+)", options, turkish, "abcIi\u0130xyz", "" };
+                yield return new object[] { "(?i:iI+)", options, turkish, "abcI\u0130ixyz", "" };
+                yield return new object[] { "(?i:[^IJKLM]I)", options, turkish, "ii\u0130i\u0131ab", "i\u0131" };
+
+                if (options != RegexOptions.None && options != RegexOptions.Compiled)
+                {
+                    // Otherwise the following tests fail (in disjoint cases), see one extracted active issue below
+
+                    // Expected answers in the invariant culture
+                    yield return new object[] { "(?i:iI+)", options, invariant, "abcIi\u0130xyz", "Ii" };               // <-- failing for None, Compiled
+                    yield return new object[] { "(?i:iI+)", options, invariant, "abcI\u0130ixyz", "" };                 // <-- failing for Compiled
+                    yield return new object[] { "(?i:iI+)", options, invariant, "abc\u0130IIxyz", "II" };               // <-- failing for Compiled
+                    yield return new object[] { "(?i:iI+)", options, invariant, "abc\u0130Iixyz", "Ii" };               // <-- failing for Compiled
+                    yield return new object[] { "(?i:[^IJKLM]I)", options, invariant, "ii\u0130i\u0131ab", "\u0130i" }; // <-- failing for None, Compiled
+
+                    // Expected answers in the Turkish culture
+                    yield return new object[] { "(?i:iI+)", options, turkish, "abc\u0130IIxyz", "\u0130II" };           // <-- failing for None, Compiled
+                    yield return new object[] { "(?i:iI+)", options, turkish, "abc\u0130\u0131Ixyz", "\u0130\u0131I" }; // <-- failing for None, Compiled
+                    yield return new object[] { "(?i:iI+)", options, turkish, "abc\u0130Iixyz", "\u0130I" };            // <-- failing for None, Compiled
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(Match_In_Different_Cultures_TestData))]
+        public void Match_In_Different_Cultures(string pattern, RegexOptions options, CultureInfo culture, string input, string match_expected)
+        {
+            Regex r = Create(pattern, culture, options)[0];
+            var match = r.Match(input);
+            Assert.Equal(match_expected, match.Value);
+        }
+
+        [ActiveIssue("Incorrect result of match in complied mode in Invariant culture, https://github.com/dotnet/runtime/issues/58956")]
```

Comment:
Maybe have a 2nd test with ActiveIssue disabling it and the commented line below uncommented. So when the bug is fixed, we just remove ActiveIssue.

Reply:
Oh never mind, I didn't see that Create was creating several. Maybe a name like CreateCompiledAndInterpretedRegexes?

Reply:
Yes, will do.